Esmf datamodels vector stream update

jedwards4b opened the pull request:
Test suite: 
Test baseline: 
Test namelist changes: 
Test status: [bit for bit, roundoff, climate changing]

Fixes [CIME Github issue #]

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

mvertens commented on src/share/streams_nuopc/dshr_strdata_mod.F90:
```diff
@@ -71,13 +70,14 @@ module dshr_strdata_mod
-     type(ESMF_FieldBundle)              :: fldbun_model                    ! stream n field bundle for model time (model grid)
-     integer                             :: ustrm                           ! index of vector u in stream
-     integer                             :: vstrm                           ! index of vector v in stream
+     integer                             :: stream_lb
```
Please add a comment to what this is. It would be good to have every attribute in these derived types commented.

mvertens commented on src/share/streams_nuopc/dshr_strdata_mod.F90:
```diff
@@ -71,13 +70,14 @@ module dshr_strdata_mod
-     integer                             :: vstrm                           ! index of vector v in stream
+     integer                             :: stream_lb
+     integer                             :: stream_ub
+     type(ESMF_Field)                    :: field_stream                    ! a field on the stream data domain
```
Can you please clarify if the field has the stream file name or the model filename. This was a huge source of confusion in the past.

mvertens commented on src/share/streams_nuopc/dshr_strdata_mod.F90:
```diff
@@ -870,48 +818,21 @@ subroutine shr_strdata_advance(sdat, ymd, tod, logunit, istr, timers, rc)
 
        enddo ! end of loop over streams
-
+#ifdef DOTHIS
```
So the vector field implementation has not been tested yet? If you have tested it - then we don't need this if-def. In fact it looks like you are handling the vector streams in another place - so why is this still here?

jedwards4b commented:
@mvertens I think that I've addressed your issues - I also created a difference file comparing test
SMS_Vnuopc_Ln11_D.f19_g17_rx1.A.cheyenne_intel to the apr26 baseline for this case.  
/glade/scratch/jedwards/SMS_Vnuopc_Ln11_D.f19_g17_rx1.A.cheyenne_intel.C.20200601_084934_gul611/diffat19800.nc
there may be some reason for concern about differences in U and V around the poles.
